src/server/admin/index.ts

```diff
@@ -0,0 +1,92 @@
+import Express from 'express'
+import { MessageType } from '../../shared/util/messages'
+import jsonMessage from '../util/json'
+import { DependencyIds, ERROR_404_PATH } from '../constants'
+import { displayHostname, ffExternalApi } from '../config'
+import assetVariant from '../../shared/util/asset-variant'
+import { container } from '../util/inversify'
+import ApiKeyAuthService from '../modules/user/services/ApiKeyAuthService'
+
+const BEARER_STRING = 'Bearer'
+const BEARER_SEPARATOR = ' '
+const apiKeyAuthService = container.get<ApiKeyAuthService>(
+  DependencyIds.apiKeyAuthService,
+)
+const router = Express.Router()
+
+/**
+ * To add guard for admin-user only api routes.
+ * */
+async function apiKeyAdminAuthMiddleware(
+  req: Express.Request,
+  res: Express.Response,
+  next: Express.NextFunction,
+) {
+  const authorizationHeader = req.headers.authorization
+  if (!authorizationHeader) {
+    res.unauthorized(jsonMessage('Authorization header is missing'))
+    return
+  }
+
+  const [bearerString, apiKey] = authorizationHeader.split(BEARER_SEPARATOR)
+  if (bearerString !== BEARER_STRING) {
+    res.unauthorized(jsonMessage('Invalid authorization header format'))
+    return
+  }
+  try {
+    const user = await apiKeyAuthService.getUserByApiKey(apiKey)
+    if (!user) {
+      res.unauthorized(jsonMessage('Invalid API Key'))
+      return
+    }
+    if (!(await apiKeyAuthService.isAdmin(user.id))) {
+      res.unauthorized(
+        jsonMessage(
+          `Email ${user.email} is not white listed`,
```

Comment:
nit: similarly "whitelisted" can be a single word (no spaces in between)

Reply:
updated in https://github.com/opengovsg/GoGovSG/pull/2213/commits/53bc68a3fd2ed17aad2f6189e42162ce9e57f832!